src/loader.rs

```diff
@@ -40,213 +40,232 @@ impl AssetLoader<Map> for TiledMapLoader {
+
+                        let mut positions = Vec::new();
+                        let mut normals = Vec::new();
+                        let mut uvs = Vec::new();
+                        let mut indices = Vec::new();
+
+                        let mut i = 0;
+                        for tile in chunk.tiles.iter().flat_map(|tiles_y| tiles_y.iter()) {
+                            if tile.tile_id < tileset_layer.tileset_guid {
+                                continue;
+                            }
+
+                            // X, Y
+                            positions.push([tile.vertex.x(), tile.vertex.y(), 0.0]);
+                            normals.push([0.0, 0.0, 1.0]);
+                            uvs.push([tile.uv.x(), tile.uv.w()]);
+
+                            // X, Y + 1
+                            positions.push([tile.vertex.x(), tile.vertex.w(), 0.0]);
+                            normals.push([0.0, 0.0, 1.0]);
+                            uvs.push([tile.uv.x(), tile.uv.y()]);
+
+                            // X + 1, Y + 1
+                            positions.push([tile.vertex.z(), tile.vertex.w(), 0.0]);
+                            normals.push([0.0, 0.0, 1.0]);
+                            uvs.push([tile.uv.z(), tile.uv.y()]);
+
+                            // X + 1, Y
+                            positions.push([tile.vertex.z(), tile.vertex.y(), 0.0]);
+                            normals.push([0.0, 0.0, 1.0]);
+                            uvs.push([tile.uv.z(), tile.uv.w()]);
+
+                            let mut new_indices = vec![i + 0, i + 2, i + 1, i + 0, i + 3, i + 2];
+                            indices.append(&mut new_indices);
+
+                            i += 4;
                         }
 
-                        // X, Y
-                        positions.push([tile.vertex.x(), tile.vertex.y(), 0.0]);
-                        normals.push([0.0, 0.0, 1.0]);
-                        uvs.push([tile.uv.x(), tile.uv.w()]);
-
-                        // X, Y + 1
-                        positions.push([tile.vertex.x(), tile.vertex.w(), 0.0]);
-                        normals.push([0.0, 0.0, 1.0]);
-                        uvs.push([tile.uv.x(), tile.uv.y()]);
-
-                        // X + 1, Y + 1
-                        positions.push([tile.vertex.z(), tile.vertex.w(), 0.0]);
-                        normals.push([0.0, 0.0, 1.0]);
-                        uvs.push([tile.uv.z(), tile.uv.y()]);
-
-                        // X + 1, Y
-                        positions.push([tile.vertex.z(), tile.vertex.y(), 0.0]);
-                        normals.push([0.0, 0.0, 1.0]);
-                        uvs.push([tile.uv.z(), tile.uv.w()]);
-
-                        let mut new_indices = vec![i + 0, i + 2, i + 1, i + 0, i + 3, i + 2];
-                        indices.append(&mut new_indices);
-
-                        i += 4;
+                        if positions.len() > 0 {
```

Comment:
Might be a better way to simplify how we map out the meshes so this kind of check isn't needed. It's needed not because each layer will attempt to add an "empty" mesh for each tileset that isn't in use.